objectherkenning_openbare_ruimte/databricks_pipelines/post_processing_pipeline/data_enrichment_step/components/clustering_detections.py

```diff
@@ -0,0 +1,200 @@
+
+    def filter_by_bounding_box_size(self, min_bbox_size: float):
+        # Calculate area for each image
+        self.df_joined = self.df_joined.withColumn("area", col("width") * col("height"))
+        self.df_joined = self.df_joined.where(col("area") > min_bbox_size)
+
+    def get_containers_coordinates(self):
+        if not self._containers_coordinates:
+            self._containers_coordinates = self._extract_containers_coordinates()
+        return self._containers_coordinates
+
+    def get_containers_coordinates_geometry(self):
+        if not self._containers_coordinates_geometry:
+            self._containers_coordinates_geometry = self._convert_coordinates_to_point()
+        return self._containers_coordinates_geometry
+
+    def _filter_objects_by_date(self, date):
+        self.detection_metadata = self.detection_metadata.filter(
+            self.detection_metadata["image_name"].like(f"%{date}%")
+        )
+
+    def _filter_objects_randomly(self):
+        self.detection_metadata = self.detection_metadata.sample(
+            False, 0.1, seed=42
+        )  # 10% sample size
+
+    def _filter_objects_by_tracking_id(self):
+        pass
+
+    def _join_frame_and_detection_metadata(self):
+
+        joined_df = self.frame_metadata.alias("fm").join(
+            self.detection_metadata.alias("dm"),
+            col("fm.image_name") == col("dm.image_name"),
+        )
+
+        columns = [
+            col("fm.gps_date").alias("detection_date"),
+            col("dm.id").alias("detection_id"),
+            col("dm.object_class"),
+            col("dm.image_name"),
+            col("dm.x_center"),
+            col("dm.y_center"),
+            col("dm.width"),
+            col("dm.height"),
+            col("dm.confidence"),
+            # col("fm.gps_lat"),
+            # col("fm.gps_lon"),
+            col("fm.gps_lat").cast("float"),
+            col("fm.gps_lon").cast("float"),
+        ]
+
+        joined_df = joined_df.select(columns)
+
+        return joined_df
+
+    def _extract_containers_coordinates(self):
+        # Collect the DataFrame rows as a list of Row objects
+        rows = self.df_joined.select("gps_lat", "gps_lon").collect()
+        # Convert the list of Row objects into a list of tuples
+        containers_coordinates = [(row["gps_lat"], row["gps_lon"]) for row in rows]
+        # containers_coordinates = [(float(row["gps_lat"]), float(row["gps_lon"])) for row in rows]
+
+        return containers_coordinates
+
+    def _convert_coordinates_to_point(self):
+        """
+        We need the containers coordinates as Point to perform distance calculations
+        """
+        containers_coordinates_geometry = [
+            Point(location) for location in self.get_containers_coordinates()
+        ]
+        return containers_coordinates_geometry
+
+    # why is this operation so complicated!?
+    def add_column(self, column_name, values):
```

Comment:
What column this is adding? It's not super clear what this function does

Reply:
the column added is the parameter "column_name". 
the values you want to add for the column is the "values" parameters. for me sounds clear

I can add the docstring and typing, is that enough?